test/e2e/util.go

```diff
@@ -679,7 +680,13 @@ func waitForNamespacesDeleted(c *client.Client, namespaces []string, timeout tim
 }
 
 func waitForServiceAccountInNamespace(c *client.Client, ns, serviceAccountName string, timeout time.Duration) error {
-	Logf("Waiting up to %v for service account %s to be provisioned in ns %s", timeout, serviceAccountName, ns)
+	w, err := c.ServiceAccounts(ns).Watch(api.SingleObject(api.ObjectMeta{Name: serviceAccountName}))
```

Comment:
We may want a variant that accepts a watch creator instead of a watch.  And have watch creation return a class of errors that are retryable (like Go's net packages do now).